src/CoordinatorClient.cc

```diff
@@ -585,6 +642,64 @@ RecoveryMasterFinishedRpc::wait()
 }
 
 /**
+ * Create a new lease or renew an existing one. This RPC is used internally by
+ * the RAMCloud client library to manage leases, which are used to ensure
+ * linearizability. It should not normally be invoked by application code.
+ *
+ * \param context
+ *      Overall information about this RAMCloud server or client.
+ * \param leaseId
+ *      Id of lease to be renewed if possible.
+ * \return
+ *      Valid ClientLease.
```

Comment:
What if an existing lease cannot be renewed, because it has expired? How is that reflected in the return information?